django_q/worker.py

```diff
@@ -88,26 +89,38 @@ def worker(
         # signal execution
         pre_execute.send(sender="django_q", func=f, task=task)
         # execute the payload
-        timer.value = timer_value  # Busy
+        timer.value = timer_value # Busy
+        if timer.value != -1:
+            timer.value += 3  # Add buffer so that guard doesn't kill the process on timeout before it gets processed
```

Comment:
I am probably missing something, but I don't see how this would cause issues for tasks with low timeouts. This doesn't increase the timeout of the function itself, it only increases the timeout of the worker being alive (so the "guard" can clean the worker up and start a new one after an exception has been raised).

When system.exit is called, it should stop the function from running and the timeout value for when that is called is not extended with extra seconds like the one above.